Gems/ROS2/Code/Source/RobotImporter/ROS2RobotImporterEditorSystemComponent.cpp

```diff
@@ -61,4 +77,118 @@ namespace ROS2
         options.toolbarIcon = ":/ROS2/ROS_import_icon.svg";
         AzToolsFramework::RegisterViewPane<RobotImporterWidget>("Robot Importer", "ROS2", options);
     }
+
+    bool ROS2RobotImporterEditorSystemComponent::GeneratePrefabFromFile(
+        const AZStd::string_view filePath, bool importAssetWithUrdf, bool useArticulation)
+    {
+        if (filePath.empty())
+        {
+            AZ_Warning("ROS2EditorSystemComponent", false, "Path provided for prefab is empty");
+            return false;
+        }
+        if (Utils::IsFileXacro(filePath))
+        {
+            AZ_Warning("ROS2EditorSystemComponent", false, "XACRO formatted files are not supported");
+            return false;
+        }
+
+        urdf::ModelInterfaceSharedPtr parsedUrdf = UrdfParser::ParseFromFile(filePath);
+        if (!parsedUrdf)
+        {
+            const auto log = UrdfParser::GetUrdfParsingLog();
+            AZ_Warning("ROS2EditorSystemComponent", false, "URDF parsing failed. Refer to %s", log.c_str());
+            return false;
+        }
+
+        auto collidersNames = Utils::GetMeshesFilenames(parsedUrdf->getRoot(), false, true);
+        auto visualNames = Utils::GetMeshesFilenames(parsedUrdf->getRoot(), true, false);
+        auto meshNames = Utils::GetMeshesFilenames(parsedUrdf->getRoot(), true, true);
+        AZStd::shared_ptr<Utils::UrdfAssetMap> urdfAssetsMapping = AZStd::make_shared<Utils::UrdfAssetMap>();
+        if (importAssetWithUrdf)
+        {
+            urdfAssetsMapping = AZStd::make_shared<Utils::UrdfAssetMap>(
+                Utils::CopyAssetForURDFAndCreateAssetMap(meshNames, filePath, collidersNames, visualNames));
+        }
+        bool allAssetProcessed = false;
+        bool assetProcessorFailed = false;
+
+        auto loopStartTime = AZStd::chrono::system_clock::now();
+
+        /* This loop waits until all of the assets are processed.
+           The urdf prefab cannot be created before all assets are processed.
+           There are three stop conditions: allAssetProcessed, assetProcessorFailed and a timeout.
+           After all asset are processed the allAssetProcessed will be set to true.
+           assetProcessorFailed will be set to true if the asset processor does not respond.
+           The time out will break the loop if assetLoopTimeout is exceed. */
+        while (!allAssetProcessed && !assetProcessorFailed)
+        {
+            auto loopTime = AZStd::chrono::system_clock::now();
+
+            if (loopTime - loopStartTime > assetLoopTimeout)
```

Comment:
While having a default assetLoopTimeout of 30 seconds is OK, I would suggest data driven the timeout via the settings registry, so that users change the default timeout.

```suggestion
            AZStd::chrono::seconds assetJobTimeout = assetLoopTimeout;
            auto settingsRegistry = AZ::SettingsRegistry::Get();
            if (AZ::s64 loopTimeoutValue{};
                settingsRegisty->Get(loopTimeoutValue, "/O3DE/ROS2/RobotImporter/AssetProcessorTimeoutInSeconds");
            {
                assetJobTimeout = AZStd::chrono::seconds(loopTimeoutValue);
            }
            if (loopTime - loopStartTime > assetLoopTimeout)
```

That way the user can then add a [.setreg](https://docs.o3de.org/docs/user-guide/settings/) to their project that overrides the asset processing timeout setting

#### <project-path>/Registry/ros2.setreg
```json
{
    "O3DE": {
        "ROS2": {
            "RobotImporter": {
                "AssetProcessorTimeoutInSeconds": 15
            }
        }
    }
}
```